test/cql-pytest/test_secondary_index.py

```diff
@@ -241,3 +241,15 @@ def test_filter_cluster_key(cql, test_keyspace):
         stmt = SimpleStatement(f"SELECT c1, c2 FROM {table} WHERE c1 = 1 and c2 = 1 ALLOW FILTERING")
         rows = cql.execute(stmt)
         assert_rows(rows, [1, 1])
+
+# Reproduces #9085.
+def test_multi_column(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int, c1 int, c2 int, r int, primary key(p,c1,c2)') as tbl:
+        cql.execute(f'CREATE INDEX ON {tbl}(r)')
+        cql.execute(f'INSERT INTO {tbl}(p, c1, c2, r) VALUES (1, 1, 1, 0)')
+        cql.execute(f'INSERT INTO {tbl}(p, c1, c2, r) VALUES (1, 1, 2, 1)')
+        cql.execute(f'INSERT INTO {tbl}(p, c1, c2, r) VALUES (1, 2, 1, 0)')
+        with pytest.raises(InvalidRequest, match="use ALLOW FILTERING"):
+            cql.execute(f'SELECT c1 FROM {tbl} WHERE p=1 AND (c1,c2)<(2,0) AND r=0')
+        # Uncomment when #6200 is fixed:
+        #assert_rows(cql.execute(f'SELECT c1 FROM {tbl} WHERE p=1 AND (c1,c2)<(2,0) AND r=0 ALLOW FILTERING'), [1])
```

Comment:
It would be slightly better to split this test into two cases - one for the invalid request, which passes, and one for the case which would only work after 6200 gets fixed, marked with `pytest.mark.xfail`. That way it will be easy to spot that the test starts passing, otherwise this commented line will probably stay in here forever, since nobody's going to remember its existence.

It's a nonblocking nitpick though, so I'll approve this series as a whole.

Reply:
Done.